### PR TITLE
add tID parameters, timezones, get_calendar(), run_screener()

### PR DESCRIPTION
**Pull request by b0b5h4rp13**
-- For times when you have a ticker id already, added tID parameters to get_quote and get_bars,
-- A trading bot needs to know the trading start and end times, especially on early holiday closings, so I added  get_calendar(), which also lets you know the last trading date, to know if it's a federal holiday once the trading start time has passed
-- run_screener() although knowing advancers, decliners, and most active are nice..  better to have a specific screen on price and pct gain to find what I am looking for (feel free to add more screener items, these are the ones I needed)

### CODE REVIEW

File: webull.py [hunk 1]
```diff
@@ -433,8 +466,15 @@ def cancel_otoco_order(self, order_id=''):
     '''
     get price quote
     '''
-    def get_quote(self, stock='') :
-        response = requests.get(urls.quotes(self.get_ticker(stock)))
+    def get_quote(self, stock=None, tId=None) :
+        if not stock is None:
+            tId = self.get_ticker(stock)
+        elif not tId is None:
+            tId = str(tId)
```
Comment (knurts) [inline webull.py]: format strings should have no problem with an integer, no need to convert to string here. 
Case not needed just to do tId = tId then.

File: webull.py [hunk 2]
```diff
@@ -433,8 +466,15 @@ def cancel_otoco_order(self, order_id=''):
     '''
     get price quote
     '''
-    def get_quote(self, stock='') :
-        response = requests.get(urls.quotes(self.get_ticker(stock)))
+    def get_quote(self, stock=None, tId=None) :
+        if not stock is None:
```
Comment (knurts) [inline webull.py]: If you do the other check first, can save an unnecessary function call in the case the caller provided both arguments.
if not tId is None:
    pass
elif not stock is None:
    tId = self.get_ticker(stock)
else:
    raise ValueError....

File: webull.py [hunk 3]
```diff
@@ -596,6 +643,17 @@ def get_bars(self, stock=None, interval='m1', count=1, extendTrading=0) :
             df.loc[datetime.fromtimestamp(int(row[0]))] = data
         return df.iloc[::-1]
 
+
+    def get_calendar(self):
+        params = {'type': 'm1', 'count': 1, 'extendTrading': 0}
```
Comment (knurts) [inline webull.py]: no comment, no headers, and hardcoding 'aapl'? is the goal just to see what the most recent market date was?

File: webull.py (hunk 1 above)
Comment (b0b5h4rp13) [reply webull.py]: Roger, thks for feedback

File: webull.py (hunk 2 above)
Comment (b0b5h4rp13) [reply webull.py]: Roger, thks for feedback

File: webull.py (hunk 3 above)
Comment (b0b5h4rp13) [reply webull.py]: Yes, I've added a header now, and let the symbol needed be passed in..  but it seems to be the only way to get market open/close times and make a guess on if it's a if its a federal holiday

File: webull.py [hunk 4]
```diff
@@ -644,14 +649,30 @@ def get_bars(self, stock=None, tId = None, interval='m1', count=1, extendTrading
...
+
         params = {'type': 'm1', 'count': 1, 'extendTrading': 0}
-        response = requests.get(urls.bars(self.get_ticker('AAPL')), params=params)
+        response = requests.get(urls.bars(tId), params=params)
         data = response.json()
         last_trade_date = datetime.fromtimestamp(int(data[0]['data'][0].split(',')[0]))
         for d in data[0]['dates']:
             if d['type'] == 'T':
-                return {'start':d['start'],  'end':d['end'], 'last trade':last_trade_date}
+                return {'market open':d['start'],  'market close':d['end'], 'last trade date':last_trade_date}
```
Comment (knurts) [inline webull.py]: It appears that data comes back in local timezone(Central), and dates comes back in Eastern (probably my weird acct configuration), is not consistent for providing back to caller.

It'd be nice to just have a 4th dict entry that was 'isOpen': True / False
something like 
True if today is on last_trade_date and >= open and < close else False

[{'data': ['1581714000,324.83,324.95,324.95,324.68,324.82,2004688,324.65'],
  'dates': [{'avgShow': 1,
             'end': '16:00:00',
             'start': '09:31:00',
             'type': 'T'}],
  'exchangeStatus': False,
  'hasMore': 1,
  'preClose': '324.82',
  'tickerId': 913256135,
  'timeZone': 'America/New_York'}]
1581714000 => 2020-02-14 15:00:00

FWIW: I wouldn't recommend using webull as the primary source for stock data. I like ally, alpaca, and iex for what I need. I know alpaca REST class has a get_clock function that users can just check .is_open (https://docs.alpaca.markets/api-documentation/api-v2/clock/)

Comment (b0b5h4rp13) [reply webull.py]: Just trying to make it complete...  I noticed time zone wasn't being added to the get_bars...  but it is a item in the response dict, so I will work on adding pytz to the mix. I am in eastern/nyc timezone, so it all looks the same to me. I love alpaca, but they don't want to do cash accts, so playing with webull now

**Pull request merged**